samples/cpp/videocapture_with_audio.cpp

```diff
@@ -0,0 +1,145 @@
+                cerr << "ERROR! Unable to open camera and microphone" << endl;
+                return -1;
+            }
+        } else {
+            // Open input file for both video and audio
+            vector<int> fileParams{ CAP_PROP_AUDIO_STREAM, 0, CAP_PROP_VIDEO_STREAM, 0, CAP_PROP_AUDIO_DATA_DEPTH, CV_16S };
+            cap.open(file, CAP_ANY, fileParams);
+            if (!cap.isOpened()) {
+                cerr << "ERROR! Unable to open file: " + file << endl;
+                return -1;
+            }
+        }
+    } else {
+        // Setup VideoCapture for video only
+        if (file.empty()) {
+            // No input file, capture video from default camera
+            cap.open(0, CAP_ANY);
+            if (!cap.isOpened()) {
+                cerr << "ERROR! Unable to open camera" << endl;
+                return -1;
+            }
+        } else {
+            // Open input video file
+            cap.open(file, CAP_ANY);
+            if (!cap.isOpened()) {
+                cerr << "ERROR! Unable to open file: " + file << endl;
+                return -1;
+            }
+        }
+    }
+
+    // Audio settings and info
+    const int audioBaseIndex = (int)cap.get(CAP_PROP_AUDIO_BASE_INDEX);
+    const int numberOfChannels = (int)cap.get(CAP_PROP_AUDIO_TOTAL_CHANNELS);
+    cout << "Audio Data Depth: " << depthToString((int)cap.get(CAP_PROP_AUDIO_DATA_DEPTH)) << endl;
+    cout << "Audio Samples Per Second: " << cap.get(CAP_PROP_AUDIO_SAMPLES_PER_SECOND) << endl;
+    cout << "Total Audio Channels: " << cap.get(CAP_PROP_AUDIO_TOTAL_CHANNELS) << endl;
+    cout << "Total Audio Streams: " << cap.get(CAP_PROP_AUDIO_TOTAL_STREAMS) << endl;
+
+    int numberOfFrames = 0;
+    int numberOfSamples = 0;
+    audioData.resize(numberOfChannels);
+
+    // Timing for sample and frame capture
+    const double cvTickFreq = getTickFrequency();
+    int64 sysTimeCurr = getTickCount();
+    int64 sysTimePrev = sysTimeCurr;
+
+    // Main loop for capturing and displaying frames
+    for (;;)
+    {
+        if (cap.grab())
+        {
+            // Retrieve video frame
+            cap.retrieve(videoFrame);
+            // Retrieve audio frames for all channels
+            for (int nCh = 0; nCh < numberOfChannels; nCh++)
+            {
+                cap.retrieve(audioFrame, audioBaseIndex + nCh);
+                if (!audioFrame.empty())
+                    audioData[nCh].push_back(audioFrame);
+                numberOfSamples += audioFrame.cols;
+                sysTimeCurr = getTickCount();
+            }
+            // Display video frame
+            if (!videoFrame.empty())
+            {
+                numberOfFrames++;
+                imshow("Video | q or esc to quit", videoFrame);
+                if (waitKey(30) >= 0)
+                    break;  // Exit loop if key is pressed
+            }
+            // Break after 10 seconds of audio capture
+            if ((sysTimeCurr - sysTimePrev) / cvTickFreq >= 10) {
```

Comment:
Please add console message, that stream is closed by timeout.

Reply:
Done